addons/web/static/src/js/core/owl_dialog.js

```diff
@@ -120,14 +120,17 @@ odoo.define('web.OwlDialog', function () {
 
         /**
          * @private
+         * @param {MouseEvent} ev
          */
-        _onBackdropClick() {
-            if (this.props.backdrop === 'static') {
-                if (this.mainButton) {
-                    this.mainButton.focus();
+        _onBackdropClick(ev) {
+            if (ev.target.classList.contains('modal')) {
```

Comment:
Can't this just be written `ev.target === ev.currentTarget` = "the clicked element is the one on which the event handler is bound"?

Also, instead of indenting the rest of the code, just invert the condition and return.